firedrake/ufl_expr.py

```diff
@@ -247,3 +247,8 @@ def FacetNormal(mesh):
     """
     mesh.init()
     return ufl.FacetNormal(mesh)
+
+
+for cls in [ufl.classes.IntValue, ufl.classes.FloatValue, ufl.classes.ComplexValue]:
+    # Make sure users never put literals in their forms.
+    setattr(cls, "__new__", lambda cls, value: firedrake.Constant(value))
```

Comment:
This breaks too much stuff: we can't be sure UFL isn't relying on these classes internally, for example.